lib/ice_nine/freezer/object.rb

```diff
@@ -16,6 +16,8 @@ class Object < self
       #
       # @return [Object]
       def self.guarded_deep_freeze(object, recursion_guard)
+        return object unless object.respond_to?(:freeze)
```

Comment:
Maybe it should be something like this instead:

```ruby
def self.guarded_deep_freeze(object, recursion_guard)
  object.freeze if object.respond_to?(:freeze)
  freeze_instance_variables(object, recursion_guard)
  object
end
```

Even if the object itself doesn't respond to `#freeze` we still probably want to try to freeze it's internal state.